gitman/git.py

```diff
@@ -39,29 +39,50 @@ def clone(type, repo, path, *,
     if name.endswith(".git"):
         name = name[:-4]
 
-    reference = os.path.join(cache, name + ".reference")
-    if not os.path.isdir(reference):
-        git('clone', '--mirror', repo, reference)
-
-    normpath = os.path.normpath(path)
-    if sparse_paths:
-        os.mkdir(normpath)
-        git('-C', normpath, 'init')
-        git('-C', normpath, 'config', 'core.sparseCheckout', 'true')
-        git('-C', normpath, 'remote', 'add', '-f', 'origin', reference)
-
-        with open("%s/%s/.git/info/sparse-checkout" %
-                  (os.getcwd(), normpath), 'w') as fd:
-            fd.writelines(sparse_paths)
-        with open("%s/%s/.git/objects/info/alternates" %
-                  (os.getcwd(), normpath), 'w') as fd:
-            fd.write("%s/objects" % reference)
-
-        # We use directly the revision requested here in order to respect,
-        # that not all repos have `master` as their default branch
-        git('-C', normpath, 'pull', 'origin', rev)
-    else:
-        git('clone', '--reference', reference, repo, os.path.normpath(path))
+    if settings.CACHE_DISABLE:
+        normpath = os.path.normpath(path)
+        if sparse_paths:
+            os.mkdir(normpath)
+            git('-C', normpath, 'init')
+            git('-C', normpath, 'config', 'core.sparseCheckout', 'true')
+            git('-C', normpath, 'remote', 'add', '-f', 'origin', repo)
+
+            with open("%s/%s/.git/info/sparse-checkout" %
+                    (os.getcwd(), normpath), 'w') as fd:
+                fd.writelines(sparse_paths)
+            with open("%s/%s/.git/objects/info/alternates" %
+                    (os.getcwd(), normpath), 'w') as fd:
+                fd.write("%s/objects" % repo)
+
+            # We use directly the revision requested here in order to respect,
+            # that not all repos have `master` as their default branch
+            git('-C', normpath, 'pull', 'origin', rev)
+        else:
+            git('clone', repo, os.path.normpath(path))
+    else: 
+        reference = os.path.join(cache, name + ".reference")
+        if not os.path.isdir(reference):
+            git('clone', '--mirror', repo, reference)
+
+        normpath = os.path.normpath(path)
+        if sparse_paths:
+            os.mkdir(normpath)
+            git('-C', normpath, 'init')
+            git('-C', normpath, 'config', 'core.sparseCheckout', 'true')
+            git('-C', normpath, 'remote', 'add', '-f', 'origin', reference)
+
+            with open("%s/%s/.git/info/sparse-checkout" %
+                    (os.getcwd(), normpath), 'w') as fd:
+                fd.writelines(sparse_paths)
+            with open("%s/%s/.git/objects/info/alternates" %
+                    (os.getcwd(), normpath), 'w') as fd:
+                fd.write("%s/objects" % reference)
```

Comment:
See if you can find a way to avoid duplicating this code. Maybe move the cache check inside the sparse paths check?

Reply:
Ok, I have found an option...